client/go/outline/iptable/stream_dialer.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptable
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+func lookupInTable[D any](table IPTable[D], address string) (foundDialer D, ok bool) {
```

Comment:
Does it ever make sense to have a `nil` entry?
Let's change this function to return a dialer only, no need for the boolean.
Same in the `table.Lookup`

Reply:
I think it doesn't make sense to add a nil entry to the table. In case of an empty dialer *config*, we would still add the default dialer to the table.

Because of that, we can treat nil response as not present, and remove the boolean